securitycenter/docs/snippets_list_assets.py

```diff
@@ -0,0 +1,137 @@
+
+    project_filter = (
+        "security_center_properties.resource_type="
+        + '"google.cloud.resourcemanager.Project"'
+    )
+    # list_assets returns an iterator.  We convert it to a list
+    # here for demonstration purposes only.  Processing each element
+    # from the iterator is recommended.
+    assets = list(client.list_assets(org_name, filter_=project_filter))
+    # [END demo_list_assets_with_filter]
+    assert len(assets) > 0
+    # [START demo_list_assets_with_filter]
+    print(assets)
+    # [END demo_list_assets_with_filter]
+
+
+def test_list_assets_with_filters_and_read_time():
+    """Demonstrate listing assets with a filter."""
+    from datetime import datetime, timedelta
+    from google.cloud import securitycenter_v1beta1 as securitycenter
+    from google.protobuf.timestamp_pb2 import Timestamp
+
+    # [START demo_list_assets_with_filter_and_time]
+    client = securitycenter.SecurityCenterClient()
+
+    # ORGANIZATION_ID is the numeric ID of the organization (e.g. 123213123121)
+    org_name = "organizations/{org_id}".format(org_id=ORGANIZATION_ID)
+
+    project_filter = (
+        "security_center_properties.resource_type="
+        + '"google.cloud.resourcemanager.Project"'
+    )
+
+    # Lists assets as of yesterday.
+    read_time = datetime.utcnow() - timedelta(days=1)
+    timestamp_proto = Timestamp()
+    timestamp_proto.FromDatetime(read_time)
+
+    # list_assets returns an iterator.  We convert it to a list
+    # here for demonstration purposes only.  Processing each element
+    # from the iterator is recommended.
+    assets = list(
+        client.list_assets(org_name, filter_=project_filter, read_time=timestamp_proto)
+    )
+    # [END demo_list_assets_with_filter_and_time]
+    assert len(assets) > 0
+    # [START demo_list_assets_with_filter_and_time]
+    print(assets)
+    # [END demo_list_assets_with_filter_and_time]
+
+
+def test_list_point_in_time_changes():
+    """Demonstrate listing assets along with their state changes."""
+    from google.cloud import securitycenter_v1beta1 as securitycenter
+    from google.protobuf.duration_pb2 import Duration
+    from datetime import timedelta
+
+    # [START demo_list_assets_changes]
+    client = securitycenter.SecurityCenterClient()
+
+    # ORGANIZATION_ID is the numeric ID of the organization (e.g. 123213123121)
+    org_name = "organizations/{org_id}".format(org_id=ORGANIZATION_ID)
+
+    project_filter = (
+        "security_center_properties.resource_type="
+        + '"google.cloud.resourcemanager.Project"'
+    )
+
+    # List assets and their state change the last 30 days
+    compare_delta = timedelta(days=30)
+    # Convert the timedelta to a Duration
+    duration_proto = Duration()
+    duration_proto.FromTimedelta(compare_delta)
+    # list_assets returns an iterator.  We convert it to a list
+    # here for demonstration purposes only.  Processing each element
+    # from the iterator is recommended.
+    assets = list(
```

Comment:
Personally I would prefer not doing a thing in a sample we wouldn't recommend. Is there a motivation to convert this to a list in the sample?

Reply:
Mostly so I can asset on size.  I changed this to use enumerate and print each index and result.  Please let me know if there is something more idiomatic.